crates/iota-grpc-api/proto/checkpoint.proto

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package iota.grpc;
+
+service CheckpointService {
+  rpc StreamCheckpoints (StreamRequest) returns (stream Checkpoint);
+  rpc GetEpochFirstCheckpointSequenceNumber (EpochRequest) returns (CheckpointSequenceNumberResponse);
+}
+
+message StreamRequest {
+  optional uint64 start_index = 1;
+  optional uint64 end_index = 2;
+  // If true, stream the full CheckpointData (not just the summary).
+  optional bool full = 3; // default: false
+}
+
+message EpochRequest {
+  uint64 epoch = 1;
+}
+
+message CheckpointSequenceNumberResponse {
+  uint64 sequence_number = 1;
+}
+
+message Checkpoint {
+  uint64 index = 1;
+  // BCS-encoded CertifiedCheckpointSummary (default) or CheckpointData (if full=true)
+  bytes data = 2;
```

Comment:
It might be valuable to add a `message BCS` type to the proto for this. You can then probably implement a generic trait to convert from/to the types instead of having to have a manual binary serialization function.

Reply:
Addressed this in [762a46c](https://github.com/iotaledger/iota/pull/7511/commits/762a46ca5c7e5e3c53a114c32c30e57c0c286e8d).